prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -643,6 +643,20 @@ void rxr_pkt_handle_receipt_recv(struct rxr_ep *ep,
 	if (!tx_entry) {
 		FI_WARN(&rxr_prov, FI_LOG_CQ,
 			"Failed to retrive the tx_entry when hadling receipt packet.\n");
+	}
+
+	tx_entry->receipt_received = true;
+	if (tx_entry->total_len == tx_entry->bytes_acked &&
+	    tx_entry->total_len != 0) {
+		/*
+		 * For long message protocol, when FI_DELIVERY_COMPLETE
+		 * is requested, we have to write tx completions
+		 * in either rxr_pkt_handle_data_send_completion()
+		 * or rxr_pkt_handle_receipt_recv() depending on which of them
+		 * is called later due to avoid accessing released
+		 * tx_entry.
+		 */
+		rxr_pkt_entry_release_rx(ep, pkt_entry);
```

Comment:
Shouldn't we  call ```rxr_cq_handle_tx_completion``` here? Also ```rxr_pkt_entry_release_rx``` should be called unconditionally.

Reply:
Comment addressed. Resolving.